controllers/articles.controller.js

```diff
@@ -0,0 +1,14 @@
+const { selectArticleById } = require("../models/articles.model");
+
+exports.getArticleById = (request, response, next) => {
+  selectArticleById(request.params)
+    .then((article) => {
+      if (article.length === 0) {
```

Comment:
instead of sending the res.status here, we should allow our model to send a promise.reject based on what comes back from the db so the error can be passed to the catch which will invoke the error handling middleware - this means you don't need to keep writing res.status(404) for example

Reply:
excellent point!  i've now fixed this, thank you!